CMakeLists.txt

```diff
@@ -1012,18 +1012,6 @@ if(NOT Python3_FOUND)
 endif(NOT Python3_FOUND)
 include(python)
 
-#-------------------------------------------------------------------
-# Check if PySCF is available
-#-------------------------------------------------------------------
-if(NOT QMC_NO_SLOW_CUSTOM_TESTING_COMMANDS)
-  test_python_module(pyscf HAVE_PYSCF)
```

Comment:
Sorry @quantumsteve I changed a bit my mind. I think we should keep checking pyscf here at high level. HAVE_PYSCF usage will expand.
Can we just replace test_python_module with check_python_reqs?

Reply:
test_python_module already sets the cached value `HAS_PYSCF_PYTHON_MODULE`. Can that be used in place of `HAVE_PYSCF`?

Reply:
I was not getting this part "duplicates a check". If HAS_PYSCF_PYTHON_MODULE has been cached by the previous run, should using check_python_reqs just skip the the actual check? `HAS_PYSCF_PYTHON_MODULE` is specific to the internal caching logic. so not good to expose it.

Reply:
I'll try that with `add_test=""`

Reply:
Maybe just put `test_name="pyscf"` and use the printout from `check_python_reqs`? Not sure that the direct printout from `check_python_reqs` is good to express the meaning of HAVE_PYSCF or not.

Reply:
If the printout is too messy, how about moving the caching logic to test_python_module?